### PR TITLE
Update topic page header UI 

### PR DESCRIPTION
**Pull request by yaelirub**
This PR updates the topic header according to designs. 
It also removes the "old" header from a filtered following stream since the filter bar already shows the topic and the user is already in the Following tab (screenshots to explain more)

 ### Filtered Following screen - removed header
Before             |  After
:-------------------------:|:-------------------------:
  |  

### Topic page header before
Follow              |  Following
:-------------------------:|:-------------------------:
  |  

### Topic page header After - Dark mode
Follow              |  Following
:-------------------------:|:-------------------------:
  |  

### Topic page header After - Light mode 
Follow              |  Following
:-------------------------:|:-------------------------:
  |  

To test:

Filtered Following:
1. Open the reader
2. Tap on filter 
3. Tap on Tags
4. Tap on a tag
5. See no header 
6. Remove filter and make sure everything works 

Topic page
1. Tap on an article tag 
2. See topic page according to new designs 
3. Make sure things look good in dark and light mode

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 1]
```diff
@@ -117,7 +117,7 @@ import WordPressFlux
         didSet {
             if tagSlug != nil {
                 // Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/5223
-                title = tagSlug
+                title = NSLocalizedString("TOPIC", comment: "Topic page title")
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:120]: Do we really want "TOPIC" here? It seems that the title is screaming at me. 😳

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 2]
```diff
@@ -618,7 +620,11 @@ import WordPressFlux
             return
         }
 
-        title = topic.title
+        if ReaderHelpers.isTopicTag(topic) {
+             title = NSLocalizedString("TOPIC", comment: "Topic page title")
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:624]: Same as above.

File: WordPress/Classes/ViewRelated/Reader/ReaderTagStreamHeader.swift [hunk 3]
```diff
@@ -31,8 +25,13 @@ import WordPressShared
 
     @objc open func configureHeader(_ topic: ReaderAbstractTopic) {
         titleLabel.text = topic.title
-        WPStyleGuide.applyReaderFollowButtonStyle(followButton)
         followButton.isSelected = topic.following
+
+        WPStyleGuide.applyReaderFollowTopicButtonStyle(followButton)
+        if #available(iOS 13, *) {
+            traitCollection.performAsCurrent { followButton.layer.borderColor = UIColor.gray(.shade30).cgColor }
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Reader/ReaderTagStreamHeader.swift:32]: Can we extract the color here to a constant?

Also, you might want to do the same logic for the `tintedFollowingIcon`. When toggline dark/light mode the icon remains with the same color:

| Light  | Dark |
| ------------- | ------------- |
| Screen Shot 2020-07-27 at 11 06 10 | Screen Shot 2020-07-27 at 11 07 37 |

File: WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift [hunk 4]
```diff
@@ -325,6 +325,49 @@ extension WPStyleGuide {
...
+        button.setImage(tintedFollowingIcon, for: .selected)
+
+        button.setTitle(followStringForDisplay, for: .normal)
+        button.setTitle(followingStringForDisplay, for: .selected)
+
+        button.layer.borderWidth = button.isSelected ? 1.0 : 0.0
+
+        // Default accessibility label and hint.
+        button.accessibilityLabel = button.isSelected ? followingStringForDisplay : followStringForDisplay
+        button.accessibilityHint = NSLocalizedString("Follows the tag.", comment: "VoiceOver accessibility hint, informing the user the button can be used to follow a tag.")
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift:368]: Can we extract that to a constant struct/enum?

File: WordPress/ColorPalette.xcassets/Gray30.colorset/Contents.json [hunk 5]
```diff
@@ -1,20 +1,38 @@
...
           "blue" : "150",
-          "green" : "145"
+          "green" : "145",
+          "red" : "142"
+        }
+      },
+      "idiom" : "universal"
+    },
+    {
+      "appearances" : [
```
Comment (leandroalonso) [inline WordPress/ColorPalette.xcassets/Gray30.colorset/Contents.json:16]: Is that change on purpose?

**Review by leandroalonso — CHANGES_REQUESTED** [review]
@yaelirub left a few comments!

It all looks good except an issue with the following icon when toggling dark/light mode.

File: WordPress/ColorPalette.xcassets/Gray30.colorset/Contents.json (hunk 5 above)
Comment (yaelirub) [reply WordPress/ColorPalette.xcassets/Gray30.colorset/Contents.json:16]: Yep. Its the dark appearance of gray30

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift (hunk 1 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift:120]: Lol was just following the designs
Screen Shot 2020-07-27 at 1 51 26 PM
but since the titles are not caps anyways, I'll update

File: WordPress/Classes/ViewRelated/Reader/ReaderTagStreamHeader.swift (hunk 3 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Reader/ReaderTagStreamHeader.swift]: Following DM discussion: I think it's weird to do a special handling for the following icon (the follow icon works as expected). For the sake of testing though I did:
``` 
traitCollection.performAsCurrent {
                let side = WPStyleGuide.fontSizeForTextStyle(.headline)
                let size = CGSize(width: side, height: side)
                let followingIcon = UIImage.gridicon(.readerFollowing, size: size)
             followButton.setImage(followingIcon.imageWithTintColor(WPStyleGuide.FollowButton.Style.followingTextColor), for: .selected)
                setNeedsDisplay()
}
``` 
but it didn't work :(

File: WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift [hunk 6]
```diff
@@ -550,4 +579,23 @@ extension WPStyleGuide {
         public static let contentTextStyle: UIFont.TextStyle = .callout
     }
 
+    public struct FollowButton {
```
[the remark this answers is not in the document]
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift:582]: you're drunk Hound

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift (hunk 1 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: Done in #14525

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift (hunk 2 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: #14525

File: WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift (hunk 4 above)
Comment (yaelirub) [reply WordPress/Classes/ViewRelated/Reader/WPStyleGuide+Reader.swift]: Done in #14525

**Comment by yaelirub** [comment]
@leandroalonso, ready for another look

File: WordPress/Classes/ViewRelated/Reader/ReaderTagStreamHeader.swift (hunk 3 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Reader/ReaderTagStreamHeader.swift]: @yaelirub I managed to make this work. :) 

Here's the patch: https://gist.github.com/leandroalonso/3ad0e6044f4cd458646851dde73a5476

File: WordPress/ColorPalette.xcassets/Gray30.colorset/Contents.json (hunk 5 above)
Comment (leandroalonso) [reply WordPress/ColorPalette.xcassets/Gray30.colorset/Contents.json:16]: I'm a little bit concerned about this change.

This adds a dark color for `gray30` but I guess this is not how the Muriel system was meant to work. Your change is perfectly fine but this remembers the issues we had in #12915 (which we reverted in #12944).

What we can do is — in the code — to use:

```
UIColor(light: .gray(.shade30), dark: .gray(.shade60)).cgColor
```

This will have the same result. :)

**Comment by leandroalonso** [comment]
@yaelirub almost there!

Just left 2 comments about toggling dark/light and an issue related with the dark color added. Let me know what you think. :)

File: WordPress/ColorPalette.xcassets/Gray30.colorset/Contents.json (hunk 5 above)
Comment (yaelirub) [reply WordPress/ColorPalette.xcassets/Gray30.colorset/Contents.json:16]: Excellent! Thank you for this!! <3

**Comment by yaelirub** [comment]
@leandroalonso , updated in : 420d42b40a6e050475b79e38e51690f2009d3692

**Review by leandroalonso — APPROVED** [review]
:shipit:
